src/components/CarListComponent/CarListComponent.tsx

```diff
@@ -0,0 +1,117 @@
+import ErrorMessage from '../ErrorMessage/ErrorMessage';
+import FilterComponent from '../FilterComponent/FilterComponent';
+import HeaderList from '../HeaderList/HeaderList';
+import LoadingComponent from '../LoadingComponent/Loading';
+import BodyList from './BodyList/BodyList';
+import { Container } from './CarListComponentStyled';
+
+interface usercars {
+  user_id: number;
+  car_id: number;
+}
+
+interface CarListComponentProps {
+  favoritesCarsArray?: usercars[];
+}
+
+const CarListComponent = ({ favoritesCarsArray }: CarListComponentProps) => {
+  const { data, loading, error, refetch } = useQuery_GetCarsQuery();
+  const [searchParam, setSearchParam] = useSearchParams();
+  useEffect(() => {
+    if (!loading) {
+      if (isValidUuid(searchParam.get('search')!)) {
+        refetch({
+          orderBy: [
+            {
+              sale_date: searchParam.get('order')
+                ? searchParam.get('order') == 'desc'
+                  ? Order_By.Desc
+                  : Order_By.Asc
+                : Order_By.Desc,
+            },
+          ],
+          where: {
+            _and: [
+              {
+                batch: {
+                  _eq: searchParam.get('search')!,
+                },
+              },
+              {
+                id: {
+                  _in:
+                    favoritesCarsArray &&
+                    favoritesCarsArray.map((carId) => carId.car_id),
+                },
+              },
+            ],
+          },
+        });
+      } else {
+        refetch({
+          orderBy: [
+            {
+              sale_date: searchParam.get('order')
+                ? searchParam.get('order') == 'desc'
+                  ? Order_By.Desc
+                  : Order_By.Asc
+                : Order_By.Desc,
+            },
+          ],
+          where: {
+            _or: [
+              {
+                title: {
+                  _iregex: searchParam.get('search')
+                    ? searchParam.get('search')
+                    : '',
+                },
+              },
+              {
+                vin: {
+                  _iregex: searchParam.get('search')
+                    ? searchParam.get('search')
+                    : '',
+                },
+              },
+            ],
+            id: {
+              _in:
+                favoritesCarsArray &&
+                favoritesCarsArray.map((carId) => carId.car_id),
+            },
+          },
+        });
+      }
+    }
+  }, [searchParam, favoritesCarsArray]);
+
+  return (
+    <Container>
+      <FilterComponent setSearchParam={setSearchParam} />
+
+      {loading ? (
+        <LoadingComponent />
+      ) : error ? (
+        <ErrorMessage message={error.message} />
+      ) : data?.cars.length == 0 ? (
+        <ErrorMessage message='No data Found' />
+      ) : (
+        <>
+          <HeaderList />
+          <BodyList carsInfoArray={data!} />
+        </>
+      )}

```

Comment:
Nested ternaries are no bueno. Hard to read and hard to maintain. See if you can find a way to avoid the need to nest these

Reply:
following the idea of render with ternary instead of && i refactored like so: 
``
if (loadingFavorites) {
    return <LoadingComponent />;
  }
  if (errorFavorites) {
    <ErrorMessage message={errorFavorites.message} />;
  }
  if (dataFavorites) {
    if (dataFavorites!.user_cars!.length == 0) {
      return <ErrorMessage message='No data Found' />;
    }
  }

  return (
    <Container>
      {dataFavorites ? (
        <CarListComponent favoritesCars={dataFavorites.user_cars} />
      ) : null}
    </Container>
  );
``